subiquity/ui/views/ssh.py

```diff
@@ -131,6 +117,86 @@ def validate_import_username(self):
                     "end with a hyphen."
                 )
 
+    def _select_ssh_import_id(self, sender, val):
+        iu = self.form.import_username
+        data = _ssh_import_data[val]
+        iu.help = _(data["help"])
+        iu.caption = _(data["caption"])
+        iu.widget.valid_char_pat = data["valid_char"]
+        iu.widget.error_invalid_char = _(data["error_invalid_char"])
+        iu.enabled = val is not None
+        self.form.pwauth.enabled = val is not None
+        # The logic here is a little tortured but the idea is that if
+        # the users switches from not importing a key to importing
+        # one, untick pwauth (but don't fiddle with the users choice
+        # if just switching between import sources), and conversely if
+        # the user is not importing a key then the box has to be
+        # ticked (and disabled).
+        if (val is None) != (self.form.ssh_import_id.value is None):
```

Comment:
This bit doesn't seem to work any more?

Reply:
Also it probably makes sense to disable the "import ssh key" widget until "install openssh-server" is checked?

Reply:
Oh, it looks like I had left this function out and never came back to it... I just did and adjusted the checkboxes status accordingly.